### PR TITLE
Add test to cover existing v3 api setting of tax_amount on line items

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Additional unit tests to support moving tax calc from api to BAO for line items

Before
----------------------------------------
Less tests, caching issue blocking tests

After
----------------------------------------
Test added, now clearing static when altering FinancialAccount, minor cleanup including comments & do not pass by ref

Technical Details
----------------------------------------

Comments
----------------------------------------

This is the test to ensure https://github.com/civicrm/civicrm-core/pull/18352 doesn't break stuff

### CODE REVIEW

File: CRM/Financial/BAO/FinancialTypeAccount.php [hunk 1]
```diff
@@ -71,6 +74,7 @@ public static function add(&$params, &$ids = NULL) {
     $financialTypeAccount->copyValues($params);
     self::validateRelationship($financialTypeAccount);
     $financialTypeAccount->save();
+    unset(Civi::$statics['CRM_Core_PseudoConstant']['taxRates']);
```
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialTypeAccount.php:77]: This is the actual change

File: CRM/Financial/BAO/FinancialTypeAccount.php [hunk 2]
```diff
@@ -52,14 +52,17 @@ public static function retrieve(&$params, &$defaults = [], &$allValues = []) {
...
+   *   Reference array contains one possible value
+   *   - entityFinancialAccount.
    *
-   * @return object
+   * @return CRM_Financial_DAO_EntityFinancialAccount
+   *
+   * @throws \CRM_Core_Exception
    */
-  public static function add(&$params, &$ids = NULL) {
+  public static function add(&$params, $ids = NULL) {
```
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialTypeAccount.php:62]: not altered, do not pass-by-ref

**Comment by eileenmcnaughton** [comment]
@seamuslee001 this is just a test + a static fix to support it

**Comment by seamuslee001** [comment]
This looks ok to me

**Pull request merged**